### PR TITLE
Use ScreenCoordsXY for MouseInput

### PR DESCRIPTION
**Pull request by tupaschoal**

### CODE REVIEW

File: src/openrct2-ui/input/MouseInput.cpp [hunk 1]
```diff
@@ -109,26 +109,27 @@ void game_handle_input()
...
+        game_handle_input_mouse(screenCoords, state & 0xFF);
     }
 
     if (_inputFlags & INPUT_FLAG_5)
     {
-        game_handle_input_mouse(x, y, state);
+        game_handle_input_mouse(screenCoords, state);
     }
-    else if (x != MONEY32_UNDEFINED)
+    else if (screenCoords.x != MONEY32_UNDEFINED)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/input/MouseInput.cpp]: This looks v. odd. We should find out if this information can be passed back in a nicer manor. Might not be appropriate to change it in this refactor though.

File: src/openrct2-ui/input/MouseInput.cpp [hunk 2]
```diff
@@ -1587,25 +1589,25 @@ void game_handle_edge_scroll()
...
+    input_scroll_viewport(ScreenCoordsXY(scrollX, scrollY));
 }
 
 bool input_test_place_object_modifier(PLACE_OBJECT_MODIFIER modifier)
 {
     return gInputPlaceObjectModifier & modifier;
 }
 
-void input_scroll_viewport(int32_t scrollX, int32_t scrollY)
+void input_scroll_viewport(ScreenCoordsXY scrollScreenCoords)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/input/MouseInput.cpp]: `Screen` is a bit superfluous in the variable name but don't really mind.

File: src/openrct2-ui/input/MouseInput.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2-ui/input/MouseInput.cpp]: This looks like dead code, no one ever assigns MONEY32_UNDEFINED to the coordinates.

Comment (tupaschoal) [reply src/openrct2-ui/input/MouseInput.cpp]: It is is very, if you think it's worth investigating, open an issue and I can follow up on that later.

Comment (ZehMatt) [reply src/openrct2-ui/input/MouseInput.cpp]: I've created the issue https://github.com/OpenRCT2/OpenRCT2/issues/10121

**Pull request merged**